lib/managers/files.js

```diff
@@ -1344,6 +1375,53 @@ Files.prototype.getRepresentationInfo = function(fileID, representationType, cal
 		})
 		.asCallback(callback);
 };
+
+/**
+ * Get the contents of a representation of a file, e.g, the binary content of an image or pdf.
+ *
+ * API Endpoint: '/files/:fileID?fields=representations'
+ * Method : GET
+ *
+ * @param {string} fileID The file ID to get the representation of
+ * @param {string} representationType The X-Rep-Hints type to request
+ * @param {Object} [options] Optional parameters
+ * @param {string} [options.assetPath] Asset path for representations with multiple files
+ * @param {Function} [callback] Passed a stream over the representation contents if successful
+ * @returns {Promise<Readable>} A promise resolving to a stream over the representation contents
+ */
+Files.prototype.getRepresentationContent = function(fileID, representationType, options, callback) {
+
+	options = Object.assign({ assetPath: '' }, options);
+
+	return this.getRepresentationInfo(fileID, representationType)
+		.then(reps => {
+
+			var repInfo = reps.entries.pop();
+			if (!repInfo) {
+				throw new Error('Could not get information for requested representation');
+			}
+
+			switch (repInfo.status.state) {
+
+			case 'success':
+			case 'viewable':
+				return repInfo.content.url_template;
+			case 'error':
+				throw new Error('Representation had error status');
+			case 'none':
+			case 'pending':
+				return pollRepresentationInfo(this.client, repInfo.info.url);
+			default:
+				throw new Error(`Unknown representation status: ${repInfo.status.state}`);
+			}
+		})
+		.then(assetURLTemplate => {
+			var url = assetURLTemplate.replace(ASSET_PATH, options.assetPath);
```

Comment:
Ideally you would use something like: https://github.com/robhicks/tiny-url-template (assuming it actually implements the RFC correctly, I just did a quick search). Using replace results in us not being able to extend the url in the future without breaking clients. (note, the preview ui kit already uses replace, but I'd prefer to at least make the correct solution an option that is explicitly considered.)

Reply:
Interesting, didn't realize this was a standard thing and not just our own replacement syntax.  I'll pull in a library to handle it.

Reply:
Thanks for being the first person to just accept my reasoning and pull in a library to do it. :)